File: app/src/main/java/com/zxcx/shitang/ui/my/selectAttention/SelectAttentionPresenter.java
package com.zxcx.shitang.ui.my.selectAttention;

import android.support.annotation.NonNull;

import com.zxcx.shitang.mvpBase.BasePresenter;

import java.util.List;

public class SelectAttentionPresenter extends BasePresenter<SelectAttentionContract.View> implements SelectAttentionContract.Presenter {

    private final SelectAttentionModel mModel;

    public SelectAttentionPresenter(@NonNull SelectAttentionContract.View view) {
        attachView(view);
        mModel = new SelectAttentionModel(this);
    }

    public void getAttentionList(){
        mModel.getAttentionList();
    }

    public void changeAttentionList(List<Integer> idList){
        mModel.changeAttentionList(idList);
    }

    @Override
    public void getDataSuccess(List<SelectAttentionBean> bean) {
        mView.getDataSuccess(bean);
    }

    @Override
    public void getDataFail(String msg) {
        mView.toastFail(msg);
    }

    @Override
    public void postSuccess() {
        mView.postSuccess();
    }

    @Override
    public void postFail(String msg) {
        mView.postFail(msg);
    }

    @Override
    public void showLoading() {
        mView.showLoading();
    }

    @Override
    public void hideLoading() {
        mView.hideLoading();
    }

    @Override
    public void startLogin() {
        mView.startLogin();
    }

    public void detachView() {
        super.detachView();
        mModel.onDestroy();
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/my/selectAttention/SelectAttentionPresenter.kt
package com.zxcx.zhizhe.ui.my.selectAttention

import com.zxcx.zhizhe.mvpBase.BasePresenter
import com.zxcx.zhizhe.ui.classify.ClassifyBean

class SelectAttentionPresenter(view: SelectAttentionContract.View) : BasePresenter<SelectAttentionContract.View>(), SelectAttentionContract.Presenter {

	private val mModel: SelectAttentionModel

	init {
		attachView(view)
		mModel = SelectAttentionModel(this)
	}

	fun getClassify() {
		mModel.getClassify()
	}

	fun changeAttentionList(idList: MutableList<Int>) {
		mModel.changeAttentionList(idList)
	}

	override fun getDataSuccess(bean: MutableList<ClassifyBean>) {
		mView.getDataSuccess(bean)
	}

	override fun getDataFail(msg: String) {
		mView.toastFail(msg)
	}

	override fun postSuccess() {
		mView.postSuccess()
	}

	override fun postFail(msg: String) {
		mView.postFail(msg)
	}

	override fun showLoading() {
		mView.showLoading()
	}

	override fun hideLoading() {
		mView.hideLoading()
	}

	override fun startLogin() {
		mView.startLogin()
	}

	override fun detachView() {
		super.detachView()
		mModel.onDestroy()
	}
}

